pkg/dogstatsd/batch.go

```diff
@@ -124,15 +132,16 @@ func (b *batcher) flush() {
 		b.flushSamples(uint32(i))
 	}
 
-	if len(b.events) > 0 {
+	if len(b.events) > 0 && b.choutEvents != nil {
```

Comment:
So if serverless agent through a mistake does end up calling `appendEvent`, we would accumulate them here forever. Would it make sense to add a panic there if the channel is nil, or maybe otherwise make sure it is never called?

Reply:
You are right, I've added this as an extra-safety but if such a mistake happened this is even worse. I'll remove second check 👍